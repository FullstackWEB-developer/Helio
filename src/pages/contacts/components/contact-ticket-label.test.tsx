import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ContactTicketLabel from './contact-ticket-label';
import Router from "react-router-dom";
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("ContactTicketLabel tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            },
            auth: {
                name: "Test"
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: [{
                label: "Test",
                parentValue: "Test"
            },{
                label: "Test",
                parentValue: "Test"
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ContactTicketLabel isDanger true correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactTicketLabel labelText={"Test"} valueText={"Test"} isDanger={true}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders ContactTicketLabel isDanger false correctly", async () => {
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactTicketLabel labelText={"Test"} valueText={"Test"} isDanger={false}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
