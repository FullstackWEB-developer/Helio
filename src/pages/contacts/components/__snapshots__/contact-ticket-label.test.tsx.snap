// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactTicketLabel tests renders ContactTicketLabel isDanger false correctly 1`] = `
<DocumentFragment>
  <span
    class="pr-1 subtitle3-small contact-ticket-label"
  >
    Test
  </span>
  <span
    class="pr-3 body3 contact-ticket-label "
  >
    Test
  </span>
</DocumentFragment>
`;

exports[`ContactTicketLabel tests renders ContactTicketLabel isDanger true correctly 1`] = `
<DocumentFragment>
  <span
    class="pr-1 subtitle3-small contact-ticket-label"
  >
    Test
  </span>
  <span
    class="pr-3 body3 contact-ticket-label text-danger"
  >
    Test
  </span>
</DocumentFragment>
`;
